src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -318,7 +318,7 @@ private void deleteCommon(String path, SMB2CreateRequest smb2CreateRequest)
         Connection connection = session.getConnection();
 
         // TODO Use Compounding
-        Future<SMB2CreateResponse> sendFuture = connection.send(smb2CreateRequest);
+        Future<SMB2CreateResponse> sendFuture = connection.send(smb2CreateRequest, session.getSigningRequired()?session.getSigningKey():null);
```

Comment:
Create a `Session.send(...)` method that takes just the packet to send. And delegate that to the `connection.send(packet, key)` call.